bin/mount-image-callback

```diff
@@ -118,14 +118,19 @@ do_umounts() {
 }
 
 cleanup() {
+	disconnect_qemu
 	if [ "${#UMOUNTS[@]}" -ne 0 ]; then
 		debug 2 "umounts: ${UMOUNTS[*]}"
-		do_umounts "${UMOUNTS[@]}"
+		do_umounts "${UMOUNTS[@]}" ||
+			{ error "failed cleaning up mounts"; return 1; }
+	fi
+	if [ -z "${TEMP_D}" -o ! -d "${TEMP_D}" ]; then
```

Comment:
i think this logic is broken, no?
if TEMP_D is empty or it is not a directory, then you remove it?

Reply:
Thanks for spotting this, my fault when converting the check from "inline" to an `if`.